modules/ding_provider/ding_provider.module

```diff
@@ -587,11 +590,16 @@ function ding_provider_get_pseudo_entity($ding_entity_id) {
   $ting_entity->ding_entity_id = $ding_entity_id;
   $ting_entity->provider_id = $local_id;
   $ting_entity->ding_entity_type = 'ding_entity';
-  $ting_entity->reply = new TingClientObject();
+  // TODO BBS-SAL: Clean out the direct property sets when done.
+  $ting_entity->reply = new TingPseudoObject();
   $ting_entity->reply->id = $ding_entity_id;
+  $ting_entity->reply->setDingId($ding_entity_id);
```

Comment:
This whole differentiating between opensearch_id (aka ding_entity_id in this function) and local_id is an opensearch thing right? Would it be possible to keep that distinction as an opensearch thing at leave it out of SAL? Eg. only have a getProviderId() and then let the opensearch implementation of TingObjectInterface have an additional getLocalId()?